pkg/collector/corechecks/ebpf/probe/ebpfcheck/probe.go

```diff
@@ -887,14 +904,16 @@ func hashMapNumberOfEntriesWithIteration(mp *ebpf.Map, buffers *entryCountBuffer
 	return numElements, nil
 }
 
-func hashMapNumberOfEntries(mp *ebpf.Map, buffers *entryCountBuffers, maxRestarts int) int64 {
+func hashMapNumberOfEntries(mp *ebpf.Map, mapid ebpf.MapID, mphCache *mapProgHelperCache, buffers *entryCountBuffers, maxRestarts int) int64 {
 	if isPerCPU(mp.Type()) {
 		return -1
 	}
 
 	var numElements int64
 	var err error
-	if ddmaps.BatchAPISupported() && mp.Type() != ebpf.HashOfMaps { // HashOfMaps doesn't work with batch API
+	if mphCache != nil && isForEachElemHelperAvailable() && mp.Type() != ebpf.HashOfMaps && mapid != 0 {
```

Comment:
I believe you are guaranteed that `mapid != 0` here.

Reply:
I was a bit afraid of tests, I removed it for now if it causes issue I will re-add it